File: src/main/java/com/example/torrent/model/api/TorrentDTO.java
package com.example.torrent.model.api;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TorrentDTO {
    private String name;
    private float progress;
    private TorrentStatusDTO status;
    private int seeds;
    private int peers;
    private int downloadRate;
    private int uploadRate;
    private long size;
}

File: src/main/js/client/models/TorrentDTO.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

export type TorrentDTO = {
    downloadRate?: number;
    name?: string;
    peers?: number;
    progress?: number;
    seeds?: number;
    size?: number;
    status?: TorrentDTO.status;
    uploadRate?: number;
};

export namespace TorrentDTO {

    export enum status {
        CHECKING_FILES = 'CHECKING_FILES',
        DOWNLOADING_METADATA = 'DOWNLOADING_METADATA',
        DOWNLOADING = 'DOWNLOADING',
        FINISHED = 'FINISHED',
        SEEDING = 'SEEDING',
        CHECKING_RESUME_DATA = 'CHECKING_RESUME_DATA',
        UNKNOWN = 'UNKNOWN',
    }


}
